packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -204,9 +204,9 @@ export function evaluateTemplate(vm: VM, html: Template): VNodes {
     if (process.env.NODE_ENV !== 'production') {
         assert.isTrue(
             isFunction(html),
-            `evaluateTemplate() second argument must be an imported template instead of ${toString(
-                html
-            )}`
+            `Invalid template returned by the render() method on ${vm}. It must return an imported template (e.g.: \`import html from "./${
```

Comment:
`vm` is an object here, so this would evaluate as `"Invalid template returned by the render() method on [object Object]"`. You could probably include [something helpful from the prototype of `vm.component` though](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/vm.ts#L178).

Reply:
`vm.tagName` is probably the simplest thing to put here.